portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -43,19 +44,29 @@ private String convertListToJson(List<Comment> comments) {
   }
 
   /**
-   * Return the value of the limit parameter if valid, 
+   * Returns the value of the limit parameter if valid, 
    * otherwise the default value.
    */
   private int getLimit(HttpServletRequest request) {
-    return getIntParameter(request, "limit", DEFAULT_COMMENT_LIMIT);
+    int limit = getIntParameter(request, "limit", DEFAULT_COMMENT_LIMIT);
```

Comment:
Looking at this method and one below I smell there is a good opportunity for `getNonNegativeIntParameter` function. :)